src/marvin/core/ChatCompletion/base/__init__.py

```diff
@@ -232,8 +279,16 @@ async def acreate(self, *args, **kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+        self = copy.deepcopy(self)
```

Comment:
This deepcopy could get really nasty if the functions or messages have complex attributes. Would a shallow copy suffice?